From: Mel Gorman <mgorman@suse.de>
Date: Tue, 10 May 2011 16:17:59 +0100
Subject: [PATCH] mm: change anon_vma linking to fix multi-process server scalability issue ia64
References: anon_vma scalability (fate#311931)
Patch-mainline: yes ()
Commit-ID: 5beb49305251e5669852ed541e8e2f2f7696c53e

Remaining chunk of
patches.suse/mm-change-anon_vma_linking-for-scalability.patch

Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 arch/ia64/perfmon/perfmon_compat.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/arch/ia64/perfmon/perfmon_compat.c b/arch/ia64/perfmon/perfmon_compat.c
index 6d484c4..6617273 100644
--- a/arch/ia64/perfmon/perfmon_compat.c
+++ b/arch/ia64/perfmon/perfmon_compat.c
@@ -675,6 +675,7 @@ int pfm_smpl_buf_alloc_compat(struct pfm_context *ctx, size_t rsize, int fd)
 		goto error_kmem;
 	}
 	memset(vma, 0, sizeof(*vma));
+	INIT_LIST_HEAD(&vma->anon_vma_chain);
 
 	/*
 	 * simulate effect of mmap()
